code/game/objects/items/flamethrower.dm

```diff
@@ -164,6 +172,7 @@
 	user.put_in_hands(ptank)
 	ptank = null
 	to_chat(user, span_notice("You remove the plasma tank from [src]!"))
+	qdel(src.GetComponent(/datum/component/bullet_intercepting))
```

Comment:
Rather than add and delete it as the tank is - add it once in init and allow the component to take an optional callback which determines if the item can or can't intercept bullets. (Then we can check for "does tank exist")